components/ShippingRates/RateTable.vue

```diff
@@ -94,19 +94,20 @@ const props = defineProps({
     default: false
   }
 })
-const emit = defineEmits(['update:modelValue', 'on-update-shipping-rates'])
+const emit = defineEmits(['update:modelValue', 'update-shipping-rates'])
 const isShippingModalOpened = ref<Boolean>(false)
 
 const hasShipping = ref(props.modelValue)
-const isModalReadOnly = ref(props.readOnly)
+
 watch(() => props.modelValue, (newValue) => {
   hasShipping.value = newValue
 })
 watch(hasShipping, (hasShipping) => {
   emit('update:modelValue', hasShipping)
 })
-const isEditMode = computed(() => !(props.readOnly))
-const isViewMode = computed(() => (props.readOnly))
+const isModalReadOnly = computed(() => (!props.isClassView))
+const isEditMode = computed(() => (props.isClassView))
+const isViewMode = computed(() => !(props.isClassView))
```

Comment:
what is different between `isModalReadOnly` and `isViewMode` ?

Reply:
turn isModalReadOnly back as ref data